src/main/java/games/strategy/util/memento/MementoImporter.java

```diff
@@ -0,0 +1,19 @@
+package games.strategy.util.memento;
+
+/**
+ * Imports a memento to its originator.
+ *
+ * @param <T> The type of the memento originator.
+ */
+public interface MementoImporter<T> {
+  /**
+   * Imports the specified memento and returns the originator.
+   *
+   * @param memento The memento to be imported; must not be {@code null}.
+   *
+   * @return The memento originator; never {@code null}.
+   *
+   * @throws MementoImportException If an error occurs while importing the memento.
+   */
+  T importMemento(Memento memento) throws MementoImportException;
```

Comment:
If a memento import exception occurrs, can a client take any actions to recover? What would expected client behavior be, try/catch log and re-throw maybe?